src/analyzer/analyzersilence.cpp

```diff
@@ -85,7 +87,26 @@ bool AnalyzerSilence::verifyFirstSound(
 }
 
 bool AnalyzerSilence::processSamples(const CSAMPLE* pIn, SINT count) {
-    std::span<const CSAMPLE> samples = mixxx::spanutil::spanFromPtrLen(pIn, count);
+    SINT numFrames = count / m_channelCount;
+
+    const CSAMPLE* pSilenceInput = pIn;
+    CSAMPLE* pMixedChannel = nullptr;
+
+    if (m_channelCount > mixxx::kAnalysisChannels) {
+        // If we have multi channel file (a stem file), we mix all the stems
+        // together in a stereo channel
```

Comment:
Can't we used the 5th channel of the STEM file with the mixed signal here, instead of doing the mix operation in the analyzer?

Reply:
No we cannot because the 5th channel is premixed with the compressor and limiter which we don't support. This shouldn't change anything for silence detection, but it will for other analyzers such as gain.

Reply:
Isn't mixMultichannelToStereo a time consuming step?

Reply:
Reading the silence analyser again, it looks like it should support multi channel indeed.